src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -2337,4 +2462,166 @@ RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
+    fogMaskSrcDescPtr->strides.wStride = 1;
+    fogMaskSrcDescPtr->strides.cStride = 1;
+    fogMaskSrcDescPtr->layout = layout;
+
+    RpptDesc fogMaskDstDesc ;
+    RpptDescPtr fogMaskDstDescPtr = &fogMaskDstDesc; //Fog resized mask destination description pointer
+    fogMaskDstDescPtr->numDims = 3;
+    fogMaskDstDescPtr->offsetInBytes = 0;
+    fogMaskDstDescPtr->dataType = RpptDataType::F32;  
+    fogMaskDstDescPtr->n = 1;
+    fogMaskDstDescPtr->c = 1;
+    fogMaskDstDescPtr->h = height;
+    fogMaskDstDescPtr->w = width;
+    fogMaskDstDescPtr->strides.nStride = height * width;
+    fogMaskDstDescPtr->strides.hStride = width;
+    fogMaskDstDescPtr->strides.wStride = 1;
+    fogMaskDstDescPtr->strides.cStride = 1;
+    fogMaskDstDescPtr->layout = layout;
+
+    RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *>(rpp::deref(rppHandle).GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem);
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + 2);
+
+    internalDstImgSizes->width = width;
+    internalDstImgSizes->height = height;
+    internalRoiTensorPtrSrc->xywhROI.xy.x = 0 ;
+    internalRoiTensorPtrSrc->xywhROI.xy.y = 0 ;
+    internalRoiTensorPtrSrc->xywhROI.roiHeight = 1080;
+    internalRoiTensorPtrSrc->xywhROI.roiWidth = 1920;
+
+    Rpp32u maskSize = FOG_MAX_HEIGHT * FOG_MAX_WIDTH;
+    Rpp32u maskSizeFloat = maskSize * sizeof(Rpp32f);
+
+    Rpp32f *d_fogAlphaMaskPtr, *d_fogIntensityMaskPtr;
+    d_fogAlphaMaskPtr =  reinterpret_cast<Rpp32f*>(internalRoiTensorPtrSrc + 4);
+    d_fogIntensityMaskPtr = reinterpret_cast<Rpp32f*>(d_fogAlphaMaskPtr + maskSize);
+
+    Rpp32f *d_dstfogAlphaMaskPtr, *d_dstfogIntensityMaskPtr;
+    d_dstfogAlphaMaskPtr = reinterpret_cast<Rpp32f*>(d_fogIntensityMaskPtr + maskSize);
+    d_dstfogIntensityMaskPtr = reinterpret_cast<Rpp32f*>(d_dstfogAlphaMaskPtr + (height * width));
+
+    CHECK_RETURN_STATUS(hipMemcpyAsync(d_fogAlphaMaskPtr, &fogAlphaMask_1920_1080[0], maskSizeFloat, hipMemcpyHostToDevice,rpp::deref(rppHandle).GetStream())); // Copying fog alpha mask from host to device asynchronously
+    hipStreamSynchronize(rpp::deref(rppHandle).GetStream());
+
+    rppt_resize_gpu(d_fogAlphaMaskPtr, fogMaskSrcDescPtr, d_dstfogAlphaMaskPtr, fogMaskDstDescPtr, internalDstImgSizes, interpolationType, internalRoiTensorPtrSrc, roiType, rppHandle); // Performing GPU resize operation on fog alpha mask
+    
+    CHECK_RETURN_STATUS(hipMemcpyAsync(d_fogIntensityMaskPtr, &fogIntensityMask_1920_1080[0], maskSizeFloat, hipMemcpyHostToDevice, rpp::deref(rppHandle).GetStream())); // Copying fog intensity mask from host to device asynchronously
+    hipStreamSynchronize(rpp::deref(rppHandle).GetStream());
+
+    rppt_resize_gpu(d_fogIntensityMaskPtr, fogMaskSrcDescPtr, d_dstfogIntensityMaskPtr, fogMaskDstDescPtr, internalDstImgSizes, interpolationType, internalRoiTensorPtrSrc, roiType, rppHandle); // Performing GPU resize operation on fog intensity mask
```

Comment:
Since you are copying to separate hip buffers for alpha and intensity
we can considering calling a single resize call with batchsize = 2, that resizes both alpha and intensity in single resize call